query/compile.go

```diff
@@ -550,6 +552,61 @@ func (c *compiledField) compileMovingAverage(args []influxql.Expr) error {
 	}
 }
 
+func (c *compiledField) compileExponentialMovingAverage(args []influxql.Expr) error {
+	if got := len(args) - 1; got < 1 || got > 3 {
+		return fmt.Errorf("invalid number of arguments for exponential_moving_average, expected at least 1 but no more than 3, got %d", got)
+	}
+
+	switch arg1 := args[1].(type) {
+	case *influxql.IntegerLiteral:
+		if arg1.Val < 1 {
+			return fmt.Errorf("exponential_moving_average period must be greater than or equal to 1")
+		}
+	default:
+		return fmt.Errorf("exponential_moving_average period must be an integer")
+	}
+
+	if len(args) >= 3 {
+		switch arg2 := args[2].(type) {
+		case *influxql.IntegerLiteral:
+			if arg2.Val < 0 && arg2.Val != -1 {
+				return fmt.Errorf("exponential_moving_average hold period must be greater than or equal to 0")
+			}
+		default:
+			return fmt.Errorf("exponential_moving_average hold period must be an integer")
+		}
+	}
+
+	if len(args) >= 4 {
+		switch arg3 := args[3].(type) {
+		case *influxql.StringLiteral:
+			switch arg3.String() {
+			case "'exponential_moving_average'":
```

Comment:
The string here should not have single quotes. That will get removed on its own. I think using a name of `exponential_moving_average` isn't the best. Maybe this should just be `exponential`?

Reply:
Following from your other comments, I'm assuming you mean if `arg3.Val` is used instead. Valid point. I probably just didn't notice `Val` when I was doing the discovery & implementation.

As for the name, yes it's long, but it makes it clear what it is. "exponential" is ambiguous. Most other technical analysis applications just refer to it as "ema", but influxdb doesn't abbreviate anything, and the function name is also "exponential_moving_average", so I kept the long name for consistency.
Perhaps support both? `ema` and `exponential_moving_average`?